cloud/aws/aws_s3.cc

```diff
@@ -250,6 +309,366 @@ class AwsS3ClientWrapper {
   std::shared_ptr<CloudRequestCallback> cloud_request_callback_;
 };
 
+namespace test {
```

Comment:
Could we move this implementation to the test file, so that we don't clutter the production binary?